doc/acl/Everflow-test-plan.md

```diff
@@ -0,0 +1,1134 @@
+
+Summary of the possible combinations:
+
+Combinations | ACL table stage: ingress | ACL table stage: egress
+-|-|-
+ACL Rule MIRROR_INGRESS_ACTION | [x] | N/A
+ACL Rule MIRROR_EGRESS_ACTION | [x] | [x]
+
+Totally there are 3 possible combinations. Not all the combinations are supported by all platforms. The actual combinations to be tested are determined the actual DUT platform.
+
+### Test configurations
+
+#### ACL table configurations
+
+New ACL tables of type MIRROR and MIRRORv6 need to be created in testing. The new ACL tables also need to set different values for their "stage" attribute.
+
+### Related **DUT** CLI commands
+
+Summary of the CLI commands that will be used for configuring DUT.
+
+#### `sonic-cfggen`
+
+It is for adding configuration for ACL tables. For example:
+* `sonic-cfggen -j <configuration_json_file> --write-to-db`: Load configuration in json format to config_db. The json file could be ACL table configuration.
+* `sonic-cfggen -d -v ACL_TABLE`: Dump current ACL_TABLE configuration from config_db.
+* `sonic-cfggen -d -v ACL_RULE`: Dump current ACL_RULE configuration from config_db.
+
+#### `acl-loader`
+
+It is for configuring ACL rules. For example:
+* `acl-loader update full <acl_rule_configuration_json_file> [--session_name=<session_name> --mirror_stage=<ingress|egress>]`: Load acl rules specified in a json file to config_db.
+
+#### `aclshow`
+This tool is for collecting ACL rule counters. For example:
+* `aclshow -a`
+
+#### `config mirror_session`
+
+This tool is for configuring everflow mirror session. For example:
+* `config mirror_session add <session_name> <src_ip> <dst_ip> <dscp> <ttl> [gre_type] [queue]`
+
+## Test structure
+
+### Overall structure
+
+The extended ansible test playbook will have below parts:
+1. Prepare some variables for testing
+2. Add everflow configuration
+3. Run everflow sub-tests
+4. Clear everflow configuration
+5. Repeat steps 1-3 for other configuration scenarios
+
+The subsequent sections will have more detailed description of part
+
+### Prepare some variables for testing
+
+Firstly, some variables need to be prepared for testing. For example, the source ports for injecting traffic. The expected destination ports for mirrored packets.
+
+### Add everflow configuration
+
+Before run the sub-tests for each scenario, the scripts need to setup by loading configurations to DUT for the scenario to be covered.
+
+There will be j2 template files for generating ACL tables and ACL rules configurations. Ansible playbook will generate ACL tables and ACL rules json configuration files to DUT based on these templates, switch capability and running topology. Then commands `sonic-cfggen` and `acl-loader` can be used for loading the configurations.
+
+Different sets of configuration files will be generated for different test scenarios:
+* IPv4
+  * ACL table: MIRROR, implicit ingress; ACL rules, MIRROR_INGRESS_ACTION
```

Comment:
'implicit ingress' - why do you need that separately? Orchagent makes table ingress by default - that should be a VS test case, which I believe exists, so no need to test this here.

Reply:
Agree. I'll remove this scenario if it is already covered by VS test case.